android/fakedapp/src/main/java/com/solana/mobilewalletadapter/fakedapp/MainViewModel.kt

```diff
@@ -456,11 +456,21 @@ class MainViewModel(application: Application) : AndroidViewModel(application) {
     private suspend fun doAuthorize(
         client: MobileWalletAdapterUseCase.Client,
         identity: MobileWalletAdapterUseCase.DappIdentity,
-        cluster: String?,
+        chain: String?,
         signInPayload: SignInWithSolana.Payload? = null
     ): MobileWalletAdapterClient.AuthorizationResult {
         val result = try {
-            client.authorize(identity, cluster, signInPayload)
+            if (uiState.value.sessionProtocolVersion == ProtocolVersion.V1) {
```

Comment:
clientlib-kt and the js libs handle this internally, but fakedapp needs to do it manually.